lib/LanguagePicker.js

```diff
@@ -101,29 +101,26 @@ LanguagePicker.prototype.newProcessor = function newProcessor() {
       }
 
       // 5. English
-      val = this.values[this.prefix ? `${this.prefix}en` : 'en'];
+      val = this.values[prefixedEnglish];
       if (val) {
         return val;
       }
 
       // 6. Any language with suffix 'Latn'
       // Only do this if the script isn't already latin;
       // otherwise, we've done it in step 3
-      for (const value of valueLangCodes) {
-        if (value && value.endsWith('Latn')) {
-          return this.values[value];
+      for (const langCode of valueLangCodes) {
+        if (langCode.endsWith('Latn')) {
+          return this.values[langCode];
         }
       }
 
       // If nothing is found:
       // - If there's a name tag, return it
       // - If there's no name tag, return first choice
-      return this.nameTag && this.values[this.nameTag] ?
-        this.values[this.nameTag] :
-        firstChoiceValue;
+      val = this.values[this.nameTag];
+      return this.nameTag && val ? val : firstChoiceValue;
```

Comment:
you are using `this.nameTag` before you check if it actually has a value.  How about
```js
if (this.nameTag) {
  val = this.values[this.nameTag];
}
return val || firstChoiceValue;
```

Reply:
ha! Missed that. 

Or, I could do 

```
return ( this.nameTag && this.values[this.namesTag] ) || firstChoiceValue;
```

Which is the same as yours, just short-formed.